relay-sampling/src/config.rs

```diff
@@ -533,150 +529,36 @@ mod tests {
 
         // At the start of the time range, sample rate is equal to the rule's initial sampling value.
         assert_eq!(
-            rule.evaluate(start, None).unwrap(),
-            SamplingValue::SampleRate { value: 1.0 }
+            decaying_fn.adjust_sample_rate(1.0, start, time_range),
+            Some(1.0)
         );
 
         // Halfway in the time range, the value is exactly between 1.0 and 0.5.
         assert_eq!(
-            rule.evaluate(halfway, None).unwrap(),
-            SamplingValue::SampleRate { value: 0.75 }
+            decaying_fn.adjust_sample_rate(1.0, halfway, time_range),
+            Some(0.75)
         );
 
         // Approaches 0.5 at the end.
         assert_eq!(
-            rule.evaluate(end, None).unwrap(),
-            SamplingValue::SampleRate {
-                // It won't go to exactly 0.5 because the time range is end-exclusive.
-                value: 0.5000028935185186
-            }
+            decaying_fn.adjust_sample_rate(1.0, end, time_range),
+            // It won't go to exactly 0.5 because the time range is end-exclusive.
+            Some(0.5000028935185186)
         );
 
         // If the end or beginning is missing, the linear decay shouldn't be run.
-        let rule_without_start = {
-            let mut rule = rule.clone();
-            rule.time_range.start = None;
-            rule
-        };
-
-        assert!(rule_without_start.evaluate(halfway, None).is_none());
-
-        let rule_without_end = {
-            let mut rule = rule.clone();
-            rule.time_range.end = None;
-            rule
-        };
-
-        assert!(rule_without_end.evaluate(halfway, None).is_none());
-    }
-
-    /// If the decayingfunction is set to `Constant` then it shouldn't adjust the sample rate.
-    #[test]
-    fn test_sample_rate_with_constant_decayingfn() {
-        let sampling_value = SamplingValue::SampleRate { value: 0.42 };
-
-        let rule = SamplingRule {
-            condition: RuleCondition::all(),
-            sampling_value,
-            ty: RuleType::Trace,
-            id: RuleId(0),
-            time_range: TimeRange {
-                start: Some(Utc.with_ymd_and_hms(1970, 10, 10, 0, 0, 0).unwrap()),
-                end: Some(Utc.with_ymd_and_hms(1970, 10, 12, 0, 0, 0).unwrap()),
-            },
-            decaying_fn: DecayingFunction::Constant,
-        };
-
-        let halfway = Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap();
-
-        assert_eq!(rule.evaluate(halfway, None), Some(sampling_value));
-    }
-
-    /// Validates the `sample_rate` method for different time range configurations.
-    /// The method should return `None` for times outside of the valid range and `Some` for times within it.
-    /// When the `start` or `end` of the range is missing, it defaults to always include times before the `end` or after the `start`, respectively.
-    #[test]
-    fn test_sample_rate_valid_time_range() {
```

Comment:
moved to `evaluation.rs` because we now check if the time range is valid in the `match_rules` function for performance reasons.